tests/test_fips.py

```diff
@@ -236,3 +269,61 @@ def test_gnutls_binary(container_per_test: ContainerData) -> None:
+)
+def test_gcrypt_binary(container_per_test: ContainerData) -> None:
+    """Check that a binary linked against gcrypt obeys the host's FIPS mode
+    setting:
+
+    - build a container image using :py:const:`DOCKERFILE_GCRYPT`
+    - run the bundled binary compiled from :file:`tests/files/fips-gcrypt-test.c` with
+      all FIPS digests and assert that it successfully calculates the message
+      digest
+    - rerun the same binary with non-FIPS digests and assert that this fails
+      with the expected error message.
+
+    """
+
+    container_per_test.connection.check_output(
+        "zypper --gpg-auto-import-keys -n ref && zypper -n in gcc  libgcrypt20 libgcrypt-devel && zypper -n clean && "
+        "gcc -Og -g3 fips-test-gcrypt.c -Wall -Wextra -Wpedantic -lgcrypt -o fips-test-gcrypt && "
+        "mv fips-test-gcrypt /bin/fips-test-gcrypt"
+    )
+
+    expected_fips_gcrypt_digests = {
+        "sha1": "c87d25a09584c040f3bfc53b570199591deb10ba648a6a6ffffdaa0badb23b8baf90b6168dd16b3a",
+        "sha224": "54655eae3d97147de34564572231c34d6d0917dd7852b5b93647fb4fe53ee97e5e0a2a4d359b5b461409dc44d9315afbc3b7d6bc5cd598e6",
+        "sha256": "4ea6a95a3a56fa6b7c1673c145198c52265fea4fe4cebef97249b39c25a733a0d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a26",
+        "sha384": "416b69644f4844065fcfe7b60f14fe07d1573420c4db2b15a6d1f4cb2b6933ffce8dcb1bce7788b962eb3d0c8d4eefc4acbfd470c22c0d95a1d10a087dc31988b9f7bfeb13be70b876a73558be664e5858d11f9459923e6e5fd838cb5708b969",
+        "sha512": "3916af571551c0c40eb19936c7ac2c090e140cad48e348dc4e9d5b6508a34ac6090daeeed3a081ce0e44c90b181987b71f09e8e0c190e5af26cd46eea724489de1c112ff908febc3b98b1693a6cd3564eaf8e5e6ca629d084d9f0eba99247cacdd72e369ff8941397c2807409ff66be64be908da17ad7b8a49a2a26c0e8086aa",
+        "sha512_224": "8b38cae68bdc4ef3336f61f68fa1b3a373abb330033a59fc51c31647f5b622b94204aa999acc71edc1f80ba7542abcdce28213a8925815a3",
+        "sha512_256": "04559e6cddea9a198f69bbd2a4c882d2fa3b9cda0edf18eb1582df42d88f7dba198bc813cc1b1bc065dba1bf261c1c9e3c9e92eb66c25638a447a744c38690ba",
+        "sha3-224": "7639f4678682e8c6b0968c894b9b58cac8cfbb6b44e0b7e99126229cdc0d4419c6e5dbc2e5fbb9fcb9dc6ea16d6b3cbca34aabb618e2f782",
+        "sha3-256": "9cd04d17df1c852d2b13536105c400e26d46ee0e865bd6d91d8683a2979fcb59265a271f568a62eb8c64e5cbedbdfd41d996303de25868af9b1892bda0bbcdfa",
+        "sha3-384": "b8b1ec43880fc0495a520434e379dce51fb0959f244c70b75555cae44dbac99f231bed131ee5a01ab8cd6e66e5da9b68ed7dc3dd68ec7a5898986cde448f567512a0f2cef69d1ed4e5fa2033950642c64f3a762b59576df4117cca5e7a52c188",
+        "sha3-512": "0bc583f093fb557e5aa6c3020cc93e9065a01600f3c8e1be17c5026799e4434c701a423ade5edae77491b7ee05c78cdc6e01af22bdc67fcd351fb1ab9a1936abc9a2f69add89697a8b935ce1bbc3d5c21f276b9e2377015003f98543bceffb8a2e1ec8d96fc99bd39d88ca6a0fd69d812362eda0937de4e3b3f0d94093a5b5cc",
+        "crc32": "02aaf9ffb095e5e3",
+        "crc32rfc1510": "0ddebf7dcb40238c",
+        "crc24rfc2440": "35a9fbd61cc4",
```

Comment:
CRC is cryptographically irrelevant and needn't be tested here
```suggestion
```

Reply:
removed